bin/tflocal

```diff
@@ -157,27 +218,74 @@ def get_region() -> str:
+    # allow configuring a custom endpoint via the environment
+    env_name = f"{service.replace('-', '_').upper().strip()}_ENDPOINT"
+    env_endpoint = os.environ.get(env_name, "").strip()
+    if env_endpoint:
+        if "://" not in env_endpoint:
+            env_endpoint = f"http://{env_endpoint}"
+        return env_endpoint
 
+    # some services need specific hostnames
+    hostname = LOCALSTACK_HOSTNAME
+    if service == "s3":
+        hostname = S3_HOSTNAME
+    elif service == "mwaa":
+        hostname = f"mwaa.{LOCALHOST_HOSTNAME}"
 
-def to_str(obj) -> bytes:
-    return obj.decode("UTF-8") if isinstance(obj, bytes) else obj
+    return f"http://{hostname}:{EDGE_PORT}"
 
 
-def determine_provider_aliases() -> list:
-    """Return a list of providers (and aliases) configured in the *.tf files (if any)"""
-    result = []
+def connect_to_service(service: str, region: str = None):
+    import boto3
+    region = region or get_region()
+    return boto3.client(
+        service, endpoint_url=get_service_endpoint(service), region_name=region,
+        aws_access_key_id="test", aws_secret_access_key="test",
+    )
+
+
+def get_or_create_bucket(bucket_name: str):
+    """Get or create a bucket in the current region."""
+    s3_client = connect_to_service("s3")
+    try:
+        return s3_client.head_bucket(Bucket=bucket_name)
+    except Exception:
+        region = s3_client.meta.region_name
+        kwargs = {}
+        if region != "us-east-1":
+            kwargs = {"CreateBucketConfiguration": {"LocationConstraint": region}}
+        return s3_client.create_bucket(Bucket=bucket_name, **kwargs)
+
+
+def get_or_create_ddb_table(table_name: str, region: str = None):
+    """Get or create a DynamoDB table with the given name."""
+    ddb_client = connect_to_service("dynamodb", region=region)
+    try:
+        return ddb_client.describe_table(TableName=table_name)
+    except Exception:
+        return ddb_client.create_table(
+            TableName=table_name, BillingMode="PAY_PER_REQUEST",
+            KeySchema=[{"AttributeName": "LockID", "KeyType": "HASH"}],
+            AttributeDefinitions=[{"AttributeName": "LockID", "AttributeType": "S"}]
+        )
+
+
+# ---
+# TF UTILS
+# ---
+
+def parse_tf_files() -> dict:
+    """Parse the local *.tf files and return a dict of <filename> -> <resource_dict>"""
+    result = {}
     for _file in glob.glob('*.tf'):
         try:
             with open(_file, 'r') as fp:
-                obj = hcl2.load(fp)
-            providers = obj.get("provider", [])
-            providers = [providers] if not isinstance(providers, list) else providers
-            aws_providers = [prov["aws"] for prov in providers if prov.get("aws")]
-            result.extend(aws_providers)
-        except Exception as e:
-            print(f"Warning: Unable to extract providers from {_file}:", e)
+                result[_file] = hcl2.load(fp)
+        except Exception:
+            pass
```

Comment:
should we return and/or log something here? it is supposed to return a dict, and there is no check for the consequential use either, e.g. 
```
tf_files = parse_tf_files()
for obj in tf_files.values():
 ....
```

Reply:
Added a log output, to inform the user about any files that cannot be parsed as HCL (currently raw `print`, we can introduce a logger at some point in the future..). 👍